api/v1alpha3/types.go

```diff
@@ -57,6 +57,10 @@ type NetworkSpec struct {
 	// APIServerLB is the configuration for the control-plane load balancer.
 	// +optional
 	APIServerLB LoadBalancerSpec `json:"apiServerLB,omitempty"`
+
+	// PrivateDNSZoneName define the zone name for the Azure Private DNS.
+	// +optional
+	PrivateDNSZoneName string `json:"privateDNSZoneName,omitempty"`
```

Comment:
Do we intend for this field to be mutable after cluster creation? If not, we should enforce immutability via webhook validation.